### ISSUE TITLE
Problem: routes with local addresses for next-hops create loops

### ISSUE THREAD

**Issue by jonnystorm** [jonnystorm/giraphe#8]
* HP Comware gives connected routes without next-hop "self" (0.0.0.0)
* Instead, connected next-hops are that router's local addresses
* This causes Layer 3 discovery to loop indefinitely

**Comment by jonnystorm** [comment]
## Example
```
       10.10.0.0   255.255.240.0              0          10.10.2.1
       10.10.2.1 255.255.255.255              0          127.0.0.1
```

**Comment by jonnystorm** [comment]
Fixed by #9.
